qiskit/visualization/latex.py

```diff
@@ -54,8 +57,7 @@ def __init__(
         initial_state=False,
         cregbundle=False,
         global_phase=None,
-        qregs=None,
-        cregs=None,
```

Comment:
Ok. First 2 seem straightforward. Will have to look at the 3rd.

Reply:
How about this as a solution. One thing to keep in mind is that the drawer classes will not work unless `_get_layered_instructions` has been called previously and the returns from that passed to the drawer.
- First off `qregs `and `cregs `are superfluous in the old and the new code. The old code still uses them, but the new code shows that everything can be done with `qubits `and `clbits`.
- Leave `qregs `and `cregs `in the drawer classes.
- Pass `qregs=None` and `cregs=None` from all the calls to the drawers from `circuit_visualization.py`.
- If a drawer detects a not None `qregs `or `cregs`, do a standard deprecation warning. The values in them will be ignored. `qregs`, `cregs` and `qubits`, `clbits `must be in sync or nothing works.
- If a drawer detects `circuit == None`, then create a new empty `QuantumCircuit` instance using `qubits `and `clbits`, and warn the user that the circuit needs to be passed in. The circuit is only used for the `find_bits` currently, so the `_data` portion is ignored.

Reply:
Yeah, that sounds like it matches what I had in mind as well.  I think there are a few other parts of the constructor that could similarly get this treatment now - `global_phase` seems like an obvious candidate - and things like `qubits` and `clbits` should eventually not be necessary, since they can also be read from the circuit.

For now, let's just do what you suggested with `qregs`, `cregs` and `global_phase`.

Reply:
Ok. I'll get the deprecations set up for these 3.